forge/ee/lib/billing/index.js

```diff
@@ -212,169 +202,144 @@ module.exports.init = async function (app) {
+            // Now do Devices
+            // const deviceBillingIds = await team.getDeviceBillingIds()
+            if (billableCounts.billingIds.devices.product) {
+                const deviceItem = existingItemsByProduct[billableCounts.billingIds.devices.product]
+                const deviceBillableCount = billableCounts.devices
                 if (deviceItem) {
-                    if (deviceItem.quantity !== billableCount) {
-                        app.log.info(`Updating team ${team.hashid} subscription device count to ${billableCount}`)
-                        const update = {
-                            quantity: billableCount,
-                            proration_behavior: prorationBehavior
-                        }
-                        try {
-                            await stripe.subscriptionItems.update(deviceItem.id, update)
-                        } catch (error) {
-                            app.log.warn(`Problem updating team ${team.hashid} subscription: ${error.message}`)
+                    // Device item already in the subscription
+                    if (deviceItem.quantity !== deviceBillableCount) {
+                        // Quantity doesn't match what we want
+                        if (deviceBillableCount === 0) {
+                            // Remove from the subscription
+                            app.log.info(`Updating team ${team.hashid} subscription: set device count to ${deviceBillableCount} - removing item`)
+                            delete deviceItem.quantity
+                            deviceItem.deleted = true
+                            itemsToUpdate.push(deviceItem)
+                        } else {
+                            // Update quantity
+                            app.log.info(`Updating team ${team.hashid} subscription: set device count to ${deviceBillableCount}`)
+                            deviceItem.quantity = deviceBillableCount
+                            itemsToUpdate.push(deviceItem)
                         }
                     }
-                } else if (billableCount > 0) {
-                    // Need to add the device item to the subscription
-                    app.log.info(`Updating team ${team.hashid} subscription device count to ${billableCount}`)
-                    const update = {
+                } else if (deviceBillableCount > 0) {
+                    // No existing device item, so add one
+                    app.log.info(`Updating team ${team.hashid} subscription device count to ${deviceBillableCount}`)
+                    itemsToUpdate.push({
+                        price: billableCounts.billingIds.devices.price,
+                        quantity: deviceBillableCount
+                    })
+                }
+            }
+            if (itemsToUpdate.length > 0) {
+                // Apply updates to the subscription
+                try {
+                    await stripe.subscriptions.update(subscription.subscription, {
```

Comment:
By adding a type hint to the `stripe` object, there is a hint that concerns me here:

![image](https://github.com/user-attachments/assets/03982a56-9de0-43b1-8497-53a136a2984e)

It states "up to 20 subscription items"

Do we need to consider guarding/splitting the updates based on `length`?

I am not familiar enough with the content of `billableCounts.instances` to know if >20 is even possible but at minimum, handling this would seem sensible?

<br>

PS, it think it is worth adding the type hint to the `stripe` object in this PR
Update [line 7](https://github.com/FlowFuse/flowfuse/pull/4621/files/aa08898e41d5c1deb34d089ca258a830aa4bc6b5#diff-4b72a2e8c48b3753d956d3c08ddcc40fbadda9b37cf8d1b31f3ca088e549c2c6R7) to 
```js
    /** @type {import('stripe').default } */
    const stripe = require('stripe')(app.config.billing.stripe.key)
```

Reply:
NOTE: to avoid confusion, the screenshot in the above was taken from `main` (not this branch) - but the subject matter is the important point i.e. `items` can only contain "up to 20" things.

Reply:
We have 5 possible billable items:
 - 1 x team plan
 - 3 x instances (s/m/l(
 - 1 x devices

No need to worry about the 20 item limit today - and no prospect of us getting near that limit in our roadmap as it stands.

Reply:
Added type hint as suggested